internal/querynodev2/segments/segment.go

```diff
@@ -123,11 +123,11 @@ func (s *baseSegment) StartPosition() *msgpb.MsgPosition {
 }
 
 func (s *baseSegment) Version() int64 {
-	return s.version
+	return s.version.Load()
 }
 
 func (s *baseSegment) UpdateVersion(version int64) {
```

Comment:
I feel like deleting the UpdateVersion method, using copy on write is better.

Reply:
This version is actually the target version maintained by QueryCoord, the segment itself is immutable.